src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -226,14 +227,36 @@ private void addOptions(Template template, Options options, String infrastructur
         if (securityGroupNames != null && !securityGroupNames.isEmpty()) {
             template.getOptions().as(AWSEC2TemplateOptions.class).securityGroupIds(securityGroupNames);
         } else {
-            usingAutoGeneratedSecurityGroupInfrastructures.add(infrastructureId);
+
             log.info(String.format("The infrastructure [%s] is using the auto-generated security group.",
-                                   infrastructureId));
+                                   infrastructure.getId()));
             // Have we defined an explicit list of ports to be be opened ?
-            Optional<int[]> portsToBeExplicitlyOpened = Optional.ofNullable(options.getPortsToOpen());
-            portsToBeExplicitlyOpened.ifPresent(integers -> template.getOptions()
-                                                                    .as(AWSEC2TemplateOptions.class)
-                                                                    .inboundPorts(integers));
+            int[] ports = options.getPortsToOpen();
+            String sgName = getAutoGeneratedSecurityGroupName(infrastructure.getId());
```

Comment:
what does mean sg? use a more meaningful name?

Reply:
ok it is SecurityGroup

Reply:
SG means "Security Group" in AWS terminology. This is a well known acronym in this context.

Reply:
security group. I'll change it.